service/matching/tasklist/task_list_manager.go

```diff
@@ -259,69 +254,51 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params AddTaskParams)
 		// request sent by history service
 		c.liveness.MarkAlive()
 	}
-	var syncMatch bool
 	e := event.E{
 		TaskListName: c.taskListID.GetName(),
 		TaskListKind: &c.taskListKind,
 		TaskListType: c.taskListID.GetType(),
 		TaskInfo:     *params.TaskInfo,
 	}
-	_, err := c.executeWithRetry(func() (interface{}, error) {
-		if err := ctx.Err(); err != nil {
-			return nil, err
-		}
-
-		domainEntry, err := c.domainCache.GetDomainByID(params.TaskInfo.DomainID)
-		if err != nil {
-			return nil, err
-		}
-
-		isForwarded := params.ForwardedFrom != ""
-
-		if _, err := domainEntry.IsActiveIn(c.clusterMetadata.GetCurrentClusterName()); err != nil {
-			// standby task, only persist when task is not forwarded from child partition
-			syncMatch = false
-			if isForwarded {
-				return &persistence.CreateTasksResponse{}, errRemoteSyncMatchFailed
-			}
-
-			r, err := c.taskWriter.appendTask(params.TaskInfo)
-			return r, err
-		}
+	domainEntry, err := c.domainCache.GetDomainByID(params.TaskInfo.DomainID)
+	if err != nil {
+		return false, err
+	}
 
+	if isActive, err := domainEntry.IsActiveIn(c.clusterMetadata.GetCurrentClusterName()); err == nil && isActive {
 		isolationGroup, err := c.getIsolationGroupForTask(ctx, params.TaskInfo)
 		if err != nil {
 			return false, err
 		}
 		// active task, try sync match first
-		syncMatch, err = c.trySyncMatch(ctx, params, isolationGroup)
+		syncMatch, err := c.trySyncMatch(ctx, params, isolationGroup)
 		if syncMatch {
 			e.EventName = "SyncMatched so not persisted"
 			event.Log(e)
-			return &persistence.CreateTasksResponse{}, err
-		}
-		if params.ActivityTaskDispatchInfo != nil {
-			return false, errRemoteSyncMatchFailed
+			return syncMatch, err
 		}
+	}
 
-		if isForwarded {
-			// forwarded from child partition - only do sync match
-			// child partition will persist the task when sync match fails
-			e.EventName = "Could not SyncMatched Forwarded Task so not persisted"
-			event.Log(e)
-			return &persistence.CreateTasksResponse{}, errRemoteSyncMatchFailed
-		}
+	if params.ActivityTaskDispatchInfo != nil {
+		return false, ErrRemoteSyncMatchFailed
+	}
 
-		e.EventName = "Task Sent to Writer"
+	if params.ForwardedFrom != "" {
```

Comment:
nit: would it be possible to put this into a method like `params.IsForwarded()` so that it's clearer what this check is indicating?